exception when changing config which loads mesh

The default rviz config has a subscription to a marker publisher, which loads a mesh.
When changing the config to another which also has the same subscription, the exception below is thrown and Rviz2 terminates.

```
[ERROR] [rviz2]: ItemIdentityException: Resource with the name file:///Users/karsten/workspace/osrf/roscon2018_ws/src/roscon2018/confbot_description/meshes/Shark_t.stl already exists. in ResourceManager::add at /Users/karsten/workspace/osrf/ros2_visualization/build/rviz_ogre_vendor/ogre-master-ca665a6-prefix/src/ogre-master-ca665a6/OgreMain/src/OgreResourceManager.cpp (line 158)
libc++abi.dylib: terminating with uncaught exception of type Ogre::ItemIdentityException: ItemIdentityException: Resource with the name file:///Users/karsten/workspace/osrf/roscon2018_ws/src/roscon2018/confbot_description/meshes/Shark_t.stl already exists. in ResourceManager::add at /Users/karsten/workspace/osrf/ros2_visualization/build/rviz_ogre_vendor/ogre-master-ca665a6-prefix/src/ogre-master-ca665a6/OgreMain/src/OgreResourceManager.cpp (line 158)
Abort trap: 6
```

--- a/rviz_rendering/include/rviz_rendering/material_manager.hpp
+++ b/rviz_rendering/include/rviz_rendering/material_manager.hpp
@@ -60,6 +60,8 @@ public:
 
   static Ogre::MaterialPtr createMaterialWithShadowsAndNoLighting(std::string name);
 
+  static void createDefaultMaterials();
+
   static void enableAlphaBlending(Ogre::MaterialPtr material, float alpha);
 
   static void enableAlphaBlending(Ogre::SceneBlendType & blending, bool & depth_write, float alpha);

--- a/rviz_rendering/src/rviz_rendering/material_manager.cpp
+++ b/rviz_rendering/src/rviz_rendering/material_manager.cpp
@@ -120,4 +120,11 @@ void MaterialManager::enableAlphaBlending(
   }
 }
 
+void MaterialManager::createDefaultMaterials()
+{
+  auto material = Ogre::MaterialManager::getSingleton().create(
+    "BaseWhiteNoLighting", "rviz_rendering");
+  material->setLightingEnabled(false);
+}
+
 }  // namespace rviz_rendering

--- a/rviz_rendering/src/rviz_rendering/mesh_loader_helpers/stl_loader.cpp
+++ b/rviz_rendering/src/rviz_rendering/mesh_loader_helpers/stl_loader.cpp
@@ -42,6 +42,8 @@
 
 #include "rviz_rendering/logging.hpp"
 
+#define ROS_PACKAGE_NAME "rviz_rendering"
+
 namespace rviz_rendering
 {
 
@@ -185,7 +187,7 @@ Ogre::MeshPtr STLLoader::toMesh(const std::string & name)
 {
   std::shared_ptr<Ogre::ManualObject> object =
     std::make_shared<Ogre::ManualObject>("the one and only");
-  object->begin("BaseWhiteNoLighting", Ogre::RenderOperation::OT_TRIANGLE_LIST);
+  object->begin("BaseWhiteNoLighting", Ogre::RenderOperation::OT_TRIANGLE_LIST, ROS_PACKAGE_NAME);
 
   unsigned int vertexCount = 0;
   for (const STLLoader::Triangle & triangle : triangles_) {
@@ -193,7 +195,8 @@ Ogre::MeshPtr STLLoader::toMesh(const std::string & name)
       // Subdivide large meshes into submeshes with at most 2004
       // vertices to prevent problems on some graphics cards.
       object->end();
-      object->begin("BaseWhiteNoLighting", Ogre::RenderOperation::OT_TRIANGLE_LIST);
+      object->begin(
+        "BaseWhiteNoLighting", Ogre::RenderOperation::OT_TRIANGLE_LIST, ROS_PACKAGE_NAME);
       vertexCount = 0;
     }
 
@@ -208,8 +211,7 @@ Ogre::MeshPtr STLLoader::toMesh(const std::string & name)
 
   object->end();
 
-  Ogre::MeshPtr mesh = object->convertToMesh(name,
-      Ogre::ResourceGroupManager::DEFAULT_RESOURCE_GROUP_NAME);
+  Ogre::MeshPtr mesh = object->convertToMesh(name, ROS_PACKAGE_NAME);
   mesh->buildEdgeList();
 
   return mesh;

--- a/rviz_rendering/src/rviz_rendering/render_system.cpp
+++ b/rviz_rendering/src/rviz_rendering/render_system.cpp
@@ -48,6 +48,7 @@
 
 #include "ament_index_cpp/get_resource.hpp"
 #include "ament_index_cpp/get_resources.hpp"
+#include "rviz_rendering/material_manager.hpp"
 #include "rviz_rendering/logging.hpp"
 #include "rviz_rendering/ogre_logging.hpp"
 #include "rviz_rendering/resource_config.hpp"
@@ -354,6 +355,7 @@ RenderSystem::setupResources()
   }
 
   addAdditionalResourcesFromAmentIndex();
+  MaterialManager::createDefaultMaterials();
 }
 
 void RenderSystem::addAdditionalResourcesFromAmentIndex() const

--- a/rviz_rendering_tests/test/mesh_loader_test.cpp
+++ b/rviz_rendering_tests/test/mesh_loader_test.cpp
@@ -148,6 +148,13 @@ TEST_F(MeshLoaderTestFixture, loading_almost_valid_stl_files_should_succed) {
   ASSERT_TRUE(rviz_rendering::loadMeshFromResource(mesh_path));
 }
 
+TEST_F(MeshLoaderTestFixture, loading_stl_mesh_twice_should_not_fail) {
+  std::string mesh_path = "package://rviz_rendering_tests/test_meshes/F2.stl";
+
+  ASSERT_TRUE(rviz_rendering::loadMeshFromResource(mesh_path));
+  ASSERT_TRUE(rviz_rendering::loadMeshFromResource(mesh_path));
+}
+
 TEST_F(MeshLoaderTestFixture, can_load_assimp_mesh_files) {
   std::string mesh_path = "package://rviz_rendering_tests/test_meshes/pr2-base.dae";
 

